Added support for Adobe AIR

Matan opened the pull request.

Commit:
Added support for Adobe AIR

kevinresol reviewed (commented):
This looks nice to me but I just have a small question: where is `air` defined?

Matan commented:
Hi, the `-D air` flag would be manually set when running a swf compile intended for complication with the AIR sdk.

I'm busy writing a unified filesystem library that supports AIR. I'm successfully running the unit tests with tink_unittest. Given the these few mods on the runner.

I understand this would be difficult for you to validate as AIR isn't a common target. Here is some proof :)

```
FileTestCase
  testConstruct: [FileTestCase.hx:141] 
    - [OK] [FileTestCase.hx:145] isNotNull(_root)
    - [OK] [FileTestCase.hx:146] isNotNull(_loremIpsumTxt)
    - [OK] [FileTestCase.hx:147] isNotNull(_loremIpsumZip)
    - [OK] [FileTestCase.hx:148] isNotNull(_dashesInFilename)
    - [OK] [FileTestCase.hx:149] isNotNull(_spacesInFilename)
    - [OK] [FileTestCase.hx:150] isNotNull(_onlyName)
    - [OK] [FileTestCase.hx:151] isNotNull(_onlyExt)
    - [OK] [FileTestCase.hx:152] isNotNull(_subFolder01)
    - [OK] [FileTestCase.hx:153] isNotNull(_subFolder02)
    - [OK] [FileTestCase.hx:154] isNotNull(_subFile01)
    - [OK] [FileTestCase.hx:155] isNotNull(_subFile02)
  testResolveAbsolutePath: [FileTestCase.hx:163] 
    - [OK] [FileTestCase.hx:174] isEqual(Path.join([cwd, _root.path]), _root.resolveAbsolutePath('').path)
    - [OK] [FileTestCase.hx:175] isEqual(Path.join([cwd, _subFolder01.path]), _subFolder01.resolveAbsolutePath('').path)
    - [OK] [FileTestCase.hx:176] isEqual(Path.join([cwd, _subFile01.path]), _subFile01.resolveAbsolutePath('').path)
  testGetParent: [FileTestCase.hx:184] 
    - [OK] [FileTestCase.hx:199] isEqual(Path.join([getCWD()]), absoluteRoot.getParent().path)
    - [OK] [FileTestCase.hx:201] isEqual(_root.path, _loremIpsumTxt.getParent().path)
    - [OK] [FileTestCase.hx:202] isEqual(_root.path, _subFolder01.getParent().path)
    - [OK] [FileTestCase.hx:203] isEqual(_subFolder01.path, _subFile01.getParent().path)
  testToString: [FileTestCase.hx:213] 
    - [OK] [FileTestCase.hx:217] isEqual(_root.path, _root.toString())
    - [OK] [FileTestCase.hx:218] isEqual(_loremIpsumTxt.path, _loremIpsumTxt.toString())
    - [OK] [FileTestCase.hx:219] isEqual(_loremIpsumZip.path, _loremIpsumZip.toString())
    - [OK] [FileTestCase.hx:220] isEqual(_dashesInFilename.path, _dashesInFilename.toString())
    - [OK] [FileTestCase.hx:221] isEqual(_spacesInFilename.path, _spacesInFilename.toString())
    - [OK] [FileTestCase.hx:222] isEqual(_onlyName.path, _onlyName.toString())
    - [OK] [FileTestCase.hx:223] isEqual(_onlyExt.path, _onlyExt.toString())
    - [OK] [FileTestCase.hx:224] isEqual(_subFolder01.path, _subFolder01.toString())
    - [OK] [FileTestCase.hx:225] isEqual(_subFolder02.path, _subFolder02.toString())
    - [OK] [FileTestCase.hx:226] isEqual(_subFile01.path, _subFile01.toString())
    - [OK] [FileTestCase.hx:227] isEqual(_subFile02.path, _subFile02.toString())
  testClone: [FileTestCase.hx:235] 
    - [OK] [FileTestCase.hx:240] isNotNull(_root.clone())
    - [OK] [FileTestCase.hx:242] isFalse(_root.clone() == _root)
    - [OK] [FileTestCase.hx:244] isEqual(_root.path, _root.clone().path)
  testPath: [FileTestCase.hx:262] 
    - [OK] [FileTestCase.hx:267] isTrue(Std.is(_root.path, String))
    - [OK] [FileTestCase.hx:270] isEqual(ROOT, _root.path)
    - [OK] [FileTestCase.hx:272] isEqual(Path.join([ROOT, LOREM_IPSUM_TXT]), _loremIpsumTxt.path)
    - [OK] [FileTestCase.hx:273] isEqual(Path.join([ROOT, LOREM_IPSUM_ZIP]), _loremIpsumZip.path)
    - [OK] [FileTestCase.hx:275] isEqual(Path.join([ROOT, DASHES_IN_FILENAME_TXT]), _dashesInFilename.path)
    - [OK] [FileTestCase.hx:276] isEqual(Path.join([ROOT, SPACES_IN_FILENAME_TXT]), _spacesInFilename.path)
    - [OK] [FileTestCase.hx:278] isEqual(Path.join([ROOT, ONLY_NAME]), _onlyName.path)
    - [OK] [FileTestCase.hx:279] isEqual(Path.join([ROOT, '.' + ONLY_EXT]), _onlyExt.path)
    - [OK] [FileTestCase.hx:281] isEqual(Path.join([ROOT, SUB_FOLDER_01]), _subFolder01.path)
    - [OK] [FileTestCase.hx:282] isEqual(Path.join([ROOT, SUB_FOLDER_02]), _subFolder02.path)
    - [OK] [FileTestCase.hx:284] isEqual(Path.join([ROOT, SUB_FOLDER_01, SUB_FILE_TXT]), _subFile01.path)
    - [OK] [FileTestCase.hx:285] isEqual(Path.join([ROOT, SUB_FOLDER_02, SUB_FILE_TXT]), _subFile02.path)
  testName: [FileTestCase.hx:297] 
    - [OK] [FileTestCase.hx:301] isEqual(ROOT.split('/').pop(), _root.name)
    - [OK] [FileTestCase.hx:303] isEqual(LOREM_IPSUM, _loremIpsumTxt.name)
    - [OK] [FileTestCase.hx:304] isEqual(LOREM_IPSUM, _loremIpsumZip.name)
    - [OK] [FileTestCase.hx:306] isEqual(DASHES_IN_FILENAME, _dashesInFilename.name)
    - [OK] [FileTestCase.hx:307] isEqual(SPACES_IN_FILENAME, _spacesInFilename.name)
    - [OK] [FileTestCase.hx:309] isEqual(ONLY_NAME, _onlyName.name)
    - [OK] [FileTestCase.hx:310] isEqual("", _onlyExt.name)
    - [OK] [FileTestCase.hx:312] isEqual(SUB_FOLDER_01, _subFolder01.name)
    - [OK] [FileTestCase.hx:313] isEqual(SUB_FOLDER_02, _subFolder02.name)
    - [OK] [FileTestCase.hx:314] isEqual(SUB_FILE, _subFile01.name)
    - [OK] [FileTestCase.hx:315] isEqual(SUB_FILE, _subFile02.name)
  testDir: [FileTestCase.hx:328] 
    - [OK] [FileTestCase.hx:332] isEqual(null, _root.dir)
    - [OK] [FileTestCase.hx:334] isEqual(ROOT, _loremIpsumTxt.dir)
    - [OK] [FileTestCase.hx:335] isEqual(ROOT, _loremIpsumZip.dir)
    - [OK] [FileTestCase.hx:337] isEqual(ROOT, _dashesInFilename.dir)
    - [OK] [FileTestCase.hx:338] isEqual(ROOT, _spacesInFilename.dir)
    - [OK] [FileTestCase.hx:340] isEqual(ROOT, _onlyName.dir)
    - [OK] [FileTestCase.hx:341] isEqual(ROOT, _onlyExt.dir)
    - [OK] [FileTestCase.hx:343] isEqual(ROOT, _subFolder01.dir)
    - [OK] [FileTestCase.hx:344] isEqual(ROOT, _subFolder02.dir)
    - [OK] [FileTestCase.hx:345] isEqual(Path.join([ROOT, SUB_FOLDER_01]), _subFile01.dir)
    - [OK] [FileTestCase.hx:346] isEqual(Path.join([ROOT, SUB_FOLDER_02]), _subFile02.dir)
  testExt: [FileTestCase.hx:358] 
    - [OK] [FileTestCase.hx:362] isEqual(null, _root.ext)
    - [OK] [FileTestCase.hx:364] isEqual(TXT, _loremIpsumTxt.ext)
    - [OK] [FileTestCase.hx:365] isEqual(ZIP, _loremIpsumZip.ext)
    - [OK] [FileTestCase.hx:367] isEqual(TXT, _dashesInFilename.ext)
    - [OK] [FileTestCase.hx:368] isEqual(TXT, _spacesInFilename.ext)
    - [OK] [FileTestCase.hx:370] isEqual(null, _onlyName.ext)
    - [OK] [FileTestCase.hx:371] isEqual(ONLY_EXT, _onlyExt.ext)
    - [OK] [FileTestCase.hx:373] isEqual(null, _subFolder01.ext)
    - [OK] [FileTestCase.hx:374] isEqual(null, _subFolder02.ext)
    - [OK] [FileTestCase.hx:375] isEqual(TXT, _subFile01.ext)
    - [OK] [FileTestCase.hx:376] isEqual(TXT, _subFile02.ext)
  testFile: [FileTestCase.hx:388] 
    - [OK] [FileTestCase.hx:392] isEqual(ROOT.split('/').pop(), _root.file)
    - [OK] [FileTestCase.hx:394] isEqual(LOREM_IPSUM_TXT, _loremIpsumTxt.file)
    - [OK] [FileTestCase.hx:395] isEqual(LOREM_IPSUM_ZIP, _loremIpsumZip.file)
    - [OK] [FileTestCase.hx:397] isEqual(DASHES_IN_FILENAME_TXT, _dashesInFilename.file)
    - [OK] [FileTestCase.hx:398] isEqual(SPACES_IN_FILENAME_TXT, _spacesInFilename.file)
    - [OK] [FileTestCase.hx:400] isEqual(ONLY_NAME, _onlyName.file)
    - [OK] [FileTestCase.hx:401] isEqual('.' + ONLY_EXT, _onlyExt.file)
    - [OK] [FileTestCase.hx:403] isEqual(SUB_FOLDER_01, _subFolder01.file)
    - [OK] [FileTestCase.hx:404] isEqual(SUB_FOLDER_02, _subFolder02.file)
    - [OK] [FileTestCase.hx:405] isEqual(SUB_FILE_TXT, _subFile01.file)
    - [OK] [FileTestCase.hx:406] isEqual(SUB_FILE_TXT, _subFile02.file)
  testIsAbsolute: [FileTestCase.hx:418] 
    - [OK] [FileTestCase.hx:425] isBool(_root.isAbsolute)
    - [OK] [FileTestCase.hx:427] isBool(_loremIpsumTxt.isAbsolute)
    - [OK] [FileTestCase.hx:428] isBool(_loremIpsumZip.isAbsolute)
    - [OK] [FileTestCase.hx:430] isBool(_dashesInFilename.isAbsolute)
    - [OK] [FileTestCase.hx:431] isBool(_spacesInFilename.isAbsolute)
    - [OK] [FileTestCase.hx:433] isBool(_onlyName.isAbsolute)
    - [OK] [FileTestCase.hx:434] isBool(_onlyExt.isAbsolute)
    - [OK] [FileTestCase.hx:436] isBool(_subFolder01.isAbsolute)
    - [OK] [FileTestCase.hx:437] isBool(_subFolder02.isAbsolute)
    - [OK] [FileTestCase.hx:438] isBool(_subFile01.isAbsolute)
    - [OK] [FileTestCase.hx:439] isBool(_subFile02.isAbsolute)
    - [OK] [FileTestCase.hx:441] isTrue(new File('/some/path').isAbsolute)
  testExists: [FileTestCase.hx:462] 
    - [OK] [FileTestCase.hx:466] isTrue(_root.exists)
    - [OK] [FileTestCase.hx:468] isTrue(_loremIpsumTxt.exists)
    - [OK] [FileTestCase.hx:469] isTrue(_loremIpsumZip.exists)
    - [OK] [FileTestCase.hx:471] isTrue(_dashesInFilename.exists)
    - [OK] [FileTestCase.hx:472] isTrue(_spacesInFilename.exists)
    - [OK] [FileTestCase.hx:474] isTrue(_onlyName.exists)
    - [OK] [FileTestCase.hx:475] isTrue(_onlyExt.exists)
    - [OK] [FileTestCase.hx:477] isTrue(_subFolder01.exists)
    - [OK] [FileTestCase.hx:478] isTrue(_subFolder02.exists)
    - [OK] [FileTestCase.hx:479] isTrue(_subFile01.exists)
    - [OK] [FileTestCase.hx:480] isTrue(_subFile02.exists)
    - [OK] [FileTestCase.hx:482] isFalse(_nonExistantFile.exists)
    - [OK] [FileTestCase.hx:483] isFalse(_nonExistantFolder.exists)
    - [OK] [FileTestCase.hx:484] isFalse(_nonExistantSubFolder.exists)
    - [OK] [FileTestCase.hx:485] isFalse(_nonExistantSubFile.exists)
  testIsDirectory: [FileTestCase.hx:497] 
    - [OK] [FileTestCase.hx:501] isTrue(_root.isDirectory)
    - [OK] [FileTestCase.hx:503] isFalse(_loremIpsumTxt.isDirectory)
    - [OK] [FileTestCase.hx:504] isFalse(_loremIpsumZip.isDirectory)
    - [OK] [FileTestCase.hx:506] isFalse(_dashesInFilename.isDirectory)
    - [OK] [FileTestCase.hx:507] isFalse(_spacesInFilename.isDirectory)
    - [OK] [FileTestCase.hx:509] isFalse(_onlyName.isDirectory)
    - [OK] [FileTestCase.hx:510] isFalse(_onlyExt.isDirectory)
    - [OK] [FileTestCase.hx:512] isTrue(_subFolder01.isDirectory)
    - [OK] [FileTestCase.hx:513] isTrue(_subFolder02.isDirectory)
    - [OK] [FileTestCase.hx:514] isFalse(_subFile01.isDirectory)
    - [OK] [FileTestCase.hx:515] isFalse(_subFile02.isDirectory)
    - [OK] [FileTestCase.hx:517] isFalse(_nonExistantFile.isDirectory)
    - [OK] [FileTestCase.hx:518] isFalse(_nonExistantFolder.isDirectory)
    - [OK] [FileTestCase.hx:519] isFalse(_nonExistantSubFolder.isDirectory)
    - [OK] [FileTestCase.hx:520] isFalse(_nonExistantSubFile.isDirectory)
  testGetDirectoryListing: [FileTestCase.hx:528] 
    - [OK] [FileTestCase.hx:534] isNotNull(files)
    - [OK] [FileTestCase.hx:535] isEqual(files.length, 8)
    - [OK] [FileTestCase.hx:549] isTrue(expected.indexOf(file.path) != -1)
    - [OK] [FileTestCase.hx:549] isTrue(expected.indexOf(file.path) != -1)
    - [OK] [FileTestCase.hx:549] isTrue(expected.indexOf(file.path) != -1)
    - [OK] [FileTestCase.hx:549] isTrue(expected.indexOf(file.path) != -1)
    - [OK] [FileTestCase.hx:549] isTrue(expected.indexOf(file.path) != -1)
    - [OK] [FileTestCase.hx:549] isTrue(expected.indexOf(file.path) != -1)
    - [OK] [FileTestCase.hx:549] isTrue(expected.indexOf(file.path) != -1)
    - [OK] [FileTestCase.hx:549] isTrue(expected.indexOf(file.path) != -1)
  testGetDirectoryListingRecursive: [FileTestCase.hx:557] 
    - [OK] [FileTestCase.hx:563] isNotNull(files)
    - [OK] [FileTestCase.hx:564] isEqual(files.length, 10)
    - [OK] [FileTestCase.hx:580] isTrue(expected.indexOf(file.path) != -1)
    - [OK] [FileTestCase.hx:580] isTrue(expected.indexOf(file.path) != -1)
    - [OK] [FileTestCase.hx:580] isTrue(expected.indexOf(file.path) != -1)
    - [OK] [FileTestCase.hx:580] isTrue(expected.indexOf(file.path) != -1)
    - [OK] [FileTestCase.hx:580] isTrue(expected.indexOf(file.path) != -1)
    - [OK] [FileTestCase.hx:580] isTrue(expected.indexOf(file.path) != -1)
    - [OK] [FileTestCase.hx:580] isTrue(expected.indexOf(file.path) != -1)
    - [OK] [FileTestCase.hx:580] isTrue(expected.indexOf(file.path) != -1)
    - [OK] [FileTestCase.hx:580] isTrue(expected.indexOf(file.path) != -1)
    - [OK] [FileTestCase.hx:580] isTrue(expected.indexOf(file.path) != -1)
  testCreateDirectory: [FileTestCase.hx:592] 
    - [OK] [FileTestCase.hx:597] isFalse(_playground.exists)
    - [OK] [FileTestCase.hx:601] isTrue(_playground.exists)
    - [OK] [FileTestCase.hx:602] isTrue(_playground.isDirectory)
    - [OK] [FileTestCase.hx:608] isFalse(subFolder01.exists)
    - [OK] [FileTestCase.hx:609] isFalse(subFolder02.exists)
    - [OK] [FileTestCase.hx:610] isFalse(subFolder03.exists)
    - [OK] [FileTestCase.hx:614] isTrue(subFolder01.exists)
    - [OK] [FileTestCase.hx:615] isTrue(subFolder02.exists)
    - [OK] [FileTestCase.hx:616] isTrue(subFolder03.exists)
    - [OK] [FileTestCase.hx:618] isTrue(subFolder01.isDirectory)
    - [OK] [FileTestCase.hx:619] isTrue(subFolder02.isDirectory)
    - [OK] [FileTestCase.hx:620] isTrue(subFolder03.isDirectory)
  testCopyTo: [FileTestCase.hx:628] 
    - [OK] [FileTestCase.hx:637] isFalse(_playground.exists)
    - [OK] [FileTestCase.hx:638] isFalse(file.exists)
    - [OK] [FileTestCase.hx:639] isFalse(subFolder.exists)
    - [OK] [FileTestCase.hx:640] isFalse(subFile.exists)
    - [OK] [FileTestCase.hx:646] isTrue(_playground.exists)
    - [OK] [FileTestCase.hx:647] isTrue(_playground.isDirectory)
    - [OK] [FileTestCase.hx:648] isTrue(file.exists)
    - [OK] [FileTestCase.hx:649] isTrue(subFolder.exists)
    - [OK] [FileTestCase.hx:650] isTrue(subFolder.isDirectory)
    - [OK] [FileTestCase.hx:651] isTrue(subFile.exists)
    - [OK] [FileTestCase.hx:670] isTrue(_playground.resolvePath(path).exists)
    - [OK] [FileTestCase.hx:670] isTrue(_playground.resolvePath(path).exists)
    - [OK] [FileTestCase.hx:670] isTrue(_playground.resolvePath(path).exists)
    - [OK] [FileTestCase.hx:670] isTrue(_playground.resolvePath(path).exists)
    - [OK] [FileTestCase.hx:670] isTrue(_playground.resolvePath(path).exists)
    - [OK] [FileTestCase.hx:670] isTrue(_playground.resolvePath(path).exists)
    - [OK] [FileTestCase.hx:670] isTrue(_playground.resolvePath(path).exists)
    - [OK] [FileTestCase.hx:670] isTrue(_playground.resolvePath(path).exists)
    - [OK] [FileTestCase.hx:670] isTrue(_playground.resolvePath(path).exists)
    - [OK] [FileTestCase.hx:670] isTrue(_playground.resolvePath(path).exists)
    - [OK] [FileTestCase.hx:672] isTrue(_playground.resolvePath(SUB_FOLDER_02).isDirectory)
  testCopyInto: [FileTestCase.hx:682] 
    - [OK] [FileTestCase.hx:691] isFalse(_playground.exists)
    - [OK] [FileTestCase.hx:692] isFalse(file.exists)
    - [OK] [FileTestCase.hx:693] isFalse(subFolder.exists)
    - [OK] [FileTestCase.hx:694] isFalse(subFile.exists)
    - [OK] [FileTestCase.hx:700] isTrue(_playground.exists)
    - [OK] [FileTestCase.hx:701] isTrue(_playground.isDirectory)
    - [OK] [FileTestCase.hx:702] isTrue(file.exists)
    - [OK] [FileTestCase.hx:703] isTrue(subFolder.exists)
    - [OK] [FileTestCase.hx:704] isTrue(subFolder.isDirectory)
    - [OK] [FileTestCase.hx:705] isTrue(subFile.exists)
    - [OK] [FileTestCase.hx:724] isTrue(_playground.resolvePath(path).exists)
    - [OK] [FileTestCase.hx:724] isTrue(_playground.resolvePath(path).exists)
    - [OK] [FileTestCase.hx:724] isTrue(_playground.resolvePath(path).exists)
    - [OK] [FileTestCase.hx:724] isTrue(_playground.resolvePath(path).exists)
    - [OK] [FileTestCase.hx:724] isTrue(_playground.resolvePath(path).exists)
    - [OK] [FileTestCase.hx:724] isTrue(_playground.resolvePath(path).exists)
    - [OK] [FileTestCase.hx:724] isTrue(_playground.resolvePath(path).exists)
    - [OK] [FileTestCase.hx:724] isTrue(_playground.resolvePath(path).exists)
    - [OK] [FileTestCase.hx:724] isTrue(_playground.resolvePath(path).exists)
    - [OK] [FileTestCase.hx:724] isTrue(_playground.resolvePath(path).exists)
    - [OK] [FileTestCase.hx:726] isTrue(_playground.resolvePath(SUB_FOLDER_02).isDirectory)
  testMoveTo: [FileTestCase.hx:736] 
    - [OK] [FileTestCase.hx:746] isFalse(fileTarget.exists)
    - [OK] [FileTestCase.hx:750] isFalse(file.exists)
    - [OK] [FileTestCase.hx:751] isTrue(fileTarget.exists)
    - [OK] [FileTestCase.hx:756] isFalse(folderTarget.exists)
    - [OK] [FileTestCase.hx:760] isFalse(folder.exists)
    - [OK] [FileTestCase.hx:761] isTrue(folderTarget.exists)
 
212 Assertions   212 Success   0 Failure   0 Error 
```

kevinresol commented:
Yeah my main concern is that the `air` flag is non standard. Because for people who use the [`air3` haxelib extern](http://lib.haxe.org/p/air3/) they will get `air3` instead.

One can always override the `Reporter` class & use their own handler in `.handle()` for this kind of needs.

@back2dos what is your opinion?

Matan commented:
I see, we can change it to `air3` instead. Also, the compiler immediately complains even when not using the changed functions, don't see how to override it.

kevinresol commented:
I pushed a commit to remove the compile-time error, so it should allow you to override. https://github.com/haxetink/tink_testrunner/commit/bfb4afaa115744a6d7510a3e5ab9bbd08622eccf

e.g.:

```haxe
class FlashReporter extends Reporter {
  override function println(v:String) flash.Lib.trace(v);
}

static function main() {
  Runner.run([...], new FlashReporter())
    .handle(function(result) untyped __global__["flash.desktop.NativeApplication"].nativeApplication.exit(result.summary().failures.length)
}
```

This should fix the problem for the time being

back2dos commented:
I have no real opinion, as I don't use AIR. 

Ideally, the air3 lib would have `-D air=3` in it's extraParams.hxml, but the world is not perfect.

My advice would be to just use `(air || air3)` for the time being.

Commit:
Merge branch 'master' of github.com:haxetink/tink_testrunner into feature/air

Commit:
Added air3 def
